internal/satellite/module/gatherer/fetcher_gatherer.go

```diff
@@ -117,3 +121,7 @@ func (f *FetcherGatherer) OutputDataChannel() <-chan *queue.SequenceEvent {
 func (f *FetcherGatherer) Ack(lastOffset event.Offset) {
 	f.runningQueue.Ack(lastOffset)
 }
+
+func (f *FetcherGatherer) DependencyInjection(modules ...module.Module) {
+	f.processor = modules[0].(processor.Processor)
```

Comment:
I can see a `[0]` directly, could you explain the DI design logic?

Reply:
please recheck codes.